internal/hns/hnsendpoint.go

```diff
@@ -10,6 +10,26 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+// EndpointState represents the states of an HNS Endpoint lifecycle.
+type EndpointState uint16
+
+// EndpointState const
+// The lifecycle of an Endpoint goes through created, attached, being shared with other containers,
+// detached, after being attached, degraded and finally destroyed.
+const (
+	Uninitialized   EndpointState = iota
+	Created         EndpointState = 1
+	Attached        EndpointState = 2
+	AttachedSharing EndpointState = 3
+	Detached        EndpointState = 4
+	Degraded        EndpointState = 5
+	Destroyed       EndpointState = 6
+)
+
+func (es EndpointState) String() string {
+	return [...]string{"Uninitialized", "Attached", "AttachedSharing", "Detached", "Degraded", "Destroyed"}[es]
+}
```

Comment:
You can also use https://pkg.go.dev/golang.org/x/tools/cmd/stringer and go generate so these stay in sync. 

It looks like it is used in a few other place in the repo, but would defer to the maintainers how this should be done